### PR TITLE
MacOS CI

### PR DESCRIPTION
**Pull request by wenleix**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #205
* __->__ #203
* #202

Differential Revision: [D34343602](https://our.internmc.facebook.com/intern/diff/D34343602)

### CODE REVIEW

**Comment by wenleix** [comment]
@wenleix has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34343602).

**Review by ejguan — APPROVED** [review]

File: .github/workflows/macos.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+          python setup.py develop
+
+      - name: Install dependencies for test
+        run: |
+          pip3 install pytest
+          pip3 install torch
+
+      - name : Run TorchArrow unit tests
+        run: |
+          echo "Skipped"
```
[the remark this answers is not in the document]
Comment (wenleix) [reply .github/workflows/macos.yml:72]: @ejguan Compiler doesn't need CPU to support AVX2? -- it just need to put AVX2 instructions in the executable~

[the remark this answers is not in the document]
Comment (ejguan) [reply .github/workflows/macos.yml:72]: Yeah. I mean we can still detect instructions set during compile time.

File: .github/workflows/macos.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    runs-on: macos-latest
+
+    steps:
+      - name: Print CPU info
+        run: sysctl -a | grep machdep.cpu
+
+      - name: Setup Python environment
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
```
[the remark this answers is not in the document]
Comment (wenleix) [reply .github/workflows/macos.yml:25]: Probably later. Since today we anyway can only just build as sanity check ...

**Pull request closed without merging**